tsdb/shard.go

```diff
@@ -284,6 +293,11 @@ func (s *Shard) WritePoints(points []models.Point) error {
 
 // DeleteSeries deletes a list of series.
 func (s *Shard) DeleteSeries(seriesKeys []string) error {
+	s.mu.RLock()
```

Comment:
The difference is that `DeleteMeasurement` needs a write lock because it's writing to `s.measurementFields`, while `DeleteSeries` here is not mutating anything protected by `s.mu`, so the read lock is fine.